src/racer/core.rs

```diff
@@ -557,11 +575,21 @@ impl IndexedSource {
     }
 
     pub fn point_to_coords(&self, point: BytePos) -> Option<Coordinate> {
-        self.cache_lineoffsets();
-        // TODO: binary search is better?
-        for (i, &range) in self.lines.borrow().iter().enumerate() {
-            if range.contains(point) {
-                return Some(Coordinate::new(i as u32 + 1, (point - range.start).0 as u32))
+        if let Some(cd) = self.cache_lineoffsets_and_get_line(point) {
+            return Some(cd);
+        }
+        let lines = self.lines.borrow();
+        // find appropriate coord using binary search
+        let (mut lo, mut hi) = (0, lines.len());
```

Comment:
Could this use `slice::binary_search_by`?

Reply:
Sounds good:)